docs/man/man3/Isotope.3
.TH "Isotope" 3 "Thu Apr 11 2013" "Version v0.1" "Doxygen" \" -*- nroff -*-
.ad l
.nh
.SH NAME
Isotope \- 
.PP
The \fBIsotope\fP represents a nuclide at some temperature\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include 'pinspec/src/Isotope\&.h'\fP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "\fBIsotope\fP (char *\fB_isotope_name\fP)"
.br
.RI "\fI\fBIsotope\fP constructor\&. \fP"
.ti -1c
.RI "virtual \fB~Isotope\fP ()"
.br
.RI "\fI\fBIsotope\fP destructor deletes arrays of cross-section values that have been assigned to this isotope\&. \fP"
.ti -1c
.RI "void \fBparseName\fP ()"
.br
.RI "\fIParse input name and set atomic number for isotope\&. \fP"
.ti -1c
.RI "void \fBmakeFissionable\fP ()"
.br
.RI "\fIInform isotope that it is fissionable\&. \fP"
.ti -1c
.RI "char * \fBgetIsotopeName\fP () const "
.br
.RI "\fIReturns the name of the of isotope\&. \fP"
.ti -1c
.RI "int \fBgetA\fP () const "
.br
.RI "\fIReturns the atomic number of this isotope\&. \fP"
.ti -1c
.RI "float \fBgetAlpha\fP () const "
.br
.RI "\fIReturns the alpha $ \alpha = ((A-1)/(A+1))^2 $ values for this isotope\&. \fP"
.ti -1c
.RI "float \fBgetTemperature\fP () const "
.br
.RI "\fIReturn the temperature in degrees Kelvin for this isotope\&. \fP"
.ti -1c
.RI "float \fBgetMuAverage\fP () const "
.br
.RI "\fIReturn the average value of the cosine of the scattering angle\&. \fP"
.ti -1c
.RI "bool \fBisFissionable\fP () const "
.br
.RI "\fIReturn whether this isotope is fissionable (true) or not (false) \fP"
.ti -1c
.RI "int \fBgetNumXSEnergies\fP (char *xs_type) const "
.br
.RI "\fIReturns the number of energies for a particular cross-section type\&. \fP"
.ti -1c
.RI "float \fBgetElasticXS\fP (float energy) const "
.br
.RI "\fIReturns the microscopic elastic scattering cross-section value for some energy\&. \fP"
.ti -1c
.RI "float \fBgetElasticXS\fP (int energy_index) const "
.br
.RI "\fIReturns the microscopic elastic cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fP"
.ti -1c
.RI "float \fBgetAbsorptionXS\fP (float energy) const "
.br
.RI "\fIReturns the microscopic absorption cross-section value for some energy\&. \fP"
.ti -1c
.RI "float \fBgetAbsorptionXS\fP (int energy_index) const "
.br
.RI "\fIReturns the microscopic absorption cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fP"
.ti -1c
.RI "float \fBgetCaptureXS\fP (float energy) const "
.br
.RI "\fIReturns the microscopic capture cross-section value for some energy\&. \fP"
.ti -1c
.RI "float \fBgetCaptureXS\fP (int energy_index) const "
.br
.RI "\fIReturns the microscopic capture cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fP"
.ti -1c
.RI "float \fBgetFissionXS\fP (float energy) const "
.br
.RI "\fIReturns the microscopic fission cross-section value for some energy\&. \fP"
.ti -1c
.RI "float \fBgetFissionXS\fP (int energy_index) const "
.br
.RI "\fIReturns the microscopic fission cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fP"
.ti -1c
.RI "float \fBgetTotalXS\fP (float energy) const "
.br
.RI "\fIReturns the microscopic total cross-section value for some energy\&. \fP"
.ti -1c
.RI "float \fBgetTotalXS\fP (int energy_index) const "
.br
.RI "\fIReturns the microscopic total cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fP"
.ti -1c
.RI "float \fBgetTransportXS\fP (int energy_index) const "
.br
.RI "\fIReturns the microscopic transport cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fP"
.ti -1c
.RI "float \fBgetTransportXS\fP (float energy) const "
.br
.RI "\fIReturns the microscopic transport cross-section value for some energy\&. \fP"
.ti -1c
.RI "bool \fBusesThermalScattering\fP ()"
.br
.RI "\fIThis method returns true if the thermal scattering distributions for this isotope are to be used when sampling outgoing collision energy\&. \fP"
.ti -1c
.RI "bool \fBisRescaled\fP () const "
.br
.RI "\fIThis method returns whether or not the \fBIsotope\fP's cross-sections have been rescaled to a uniform lethargy grid\&. \fP"
.ti -1c
.RI "int \fBgetEnergyGridIndex\fP (float energy) const "
.br
.RI "\fIThis method returns the index for a certain energy (eV) into the \fBIsotope\fP's uniform lethargy grid\&. \fP"
.ti -1c
.RI "void \fBretrieveXSEnergies\fP (float *energies, int num_xs, char *xs_type) const "
.br
.RI "\fIFills an array with cross-section energy values\&. \fP"
.ti -1c
.RI "void \fBretrieveXS\fP (float *xs, int num_xs, char *xs_type) const "
.br
.RI "\fIFills an array with microscopic cross-section values\&. \fP"
.ti -1c
.RI "void \fBsetElasticXS\fP (double *energies, int num_energies, double *elastic_xs, int num_xs)"
.br
.RI "\fISets the elastic cross-section data for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetCaptureXS\fP (double *energies, int num_energies, double *capture_xs, int num_xs)"
.br
.RI "\fISets the capture cross-section data for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetFissionXS\fP (double *energies, int num_energies, double *fission_xs, int num_xs)"
.br
.RI "\fISets the fission cross-section data for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetMultigroupElasticXS\fP (double *energies, int num_energies, double *elastic_xs, int num_xs)"
.br
.RI "\fISets the multigroup elastic cross-section data for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetMultigroupCaptureXS\fP (double *energies, int num_energies, double *capture_xs, int num_xs)"
.br
.RI "\fISets the multigroup capture cross-section data for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetMultigroupFissionXS\fP (double *energies, int num_energies, double *fission_xs, int num_xs)"
.br
.RI "\fISets the multigroup fission cross-section data for this isotope\&. \fP"
.ti -1c
.RI "void \fBloadXS\fP (char *xs_type)"
.br
.RI "\fILoad the ENDF cross-section data for a particular cross-section from an ASCII file into the appropriate array for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetA\fP (int A)"
.br
.RI "\fISet the atomic number and update alpha, eta and rho\&. \fP"
.ti -1c
.RI "void \fBsetTemperature\fP (float T)"
.br
.RI "\fISet the temperature of the isotope in degrees Kelvin\&. \fP"
.ti -1c
.RI "void \fBneglectThermalScattering\fP ()"
.br
.RI "\fIInforms isotope not to use thermal scattering to sample outgoing collision energies\&. \fP"
.ti -1c
.RI "void \fBuseThermalScattering\fP ()"
.br
.RI "\fIInforms isotope to use thermal scattering to sample outgoing collision energies\&. \fP"
.ti -1c
.RI "\fBIsotope\fP * \fBclone\fP ()"
.br
.RI "\fIThis method clones a given \fBIsotope\fP class object by executing a deep copy of all of the \fBIsotope\fP's class attributes and giving them to a new \fBIsotope\fP class object\&. \fP"
.ti -1c
.RI "void \fBsampleCollisionType\fP (\fBneutron\fP *\fBneutron\fP)"
.br
.RI "\fIDetermines a random collision type based on the values of each of the isotope's cross-section values at a given enery\&. \fP"
.ti -1c
.RI "float \fBgetDistanceTraveled\fP (\fBneutron\fP *\fBneutron\fP)"
.br
.RI "\fIFor a given neutron, this method samples a distance traveled to next collision\&. \fP"
.ti -1c
.RI "void \fBcollideNeutron\fP (\fBneutron\fP *\fBneutron\fP)"
.br
.RI "\fIFor a given energy, this method samples a collision type, updates the neutron's outgoing energy and kills the neutron if it sampled absorption or leakage\&. \fP"
.ti -1c
.RI "float \fBgetThermalScatteringEnergy\fP (float energy)"
.br
.RI "\fIFor a given neutron energy (eV) in a scattering collision, this function returns the outgoing energy in eV, $ E' $, for the collision based on its thermal scattering distributions\&. \fP"
.ti -1c
.RI "int \fBgetNumThermalCDFs\fP ()"
.br
.RI "\fIReturns the number of thermal scattering CDFs\&. \fP"
.ti -1c
.RI "int \fBgetNumThermalCDFBins\fP ()"
.br
.RI "\fIReturns the number of energy bins for each thermal scattering CDF\&. \fP"
.ti -1c
.RI "void \fBretrieveThermalCDFs\fP (float *cdfs, int num_values)"
.br
.RI "\fILoads an input array with the values for each of the isotope's thermal CDFs\&. \fP"
.ti -1c
.RI "void \fBretrieveThermalPDFs\fP (float *pdfs, int num_values)"
.br
.RI "\fILoads an input array with the energies for each of the isotope's thermal PDFs\&. \fP"
.ti -1c
.RI "void \fBretrieveEtokT\fP (float *E_to_kT, int num_cdfs)"
.br
.RI "\fILoads an input array with the $ \frac{E}{kT} $ values for each thermal scattering CDF\&. \fP"
.ti -1c
.RI "void \fBretrieveEprimeToE\fP (float *Eprime_to_E, int num_bins)"
.br
.RI "\fILoads an input array with the $ \frac{E'}{E} $ values for each thermal scattering CDF\&. \fP"
.in -1c
.SS "Private Member Functions"

.in +1c
.ti -1c
.RI "void \fBloadXS\fP ()"
.br
.RI "\fILoad the ENDF cross-section data from ASCII files into arrays for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetElasticXS\fP (float *elastic_xs, float *elastic_xs_energies, int num_elastic_xs)"
.br
.RI "\fISet the elastic cross-section for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetCaptureXS\fP (float *capture_xs, float *capture_xs_energies, int num_capture_xs)"
.br
.RI "\fISet the capture cross-section for this isotope\&. \fP"
.ti -1c
.RI "void \fBsetFissionXS\fP (float *fission_xs, float *fission_xs_energies, int num_fission_xs)"
.br
.RI "\fISet the fission cross-section for this isotope\&. \fP"
.ti -1c
.RI "void \fBrescaleXS\fP (float start_energy, float end_energy, int num_energies)"
.br
.RI "\fIRescales all of the isotope's cross-sections onto a uniform lethargy grid\&. \fP"
.ti -1c
.RI "void \fBgenerateAbsorptionXS\fP (float start_energy, float end_energy, int num_energies)"
.br
.RI "\fIComputes the microscopic absorption cross-section from the isotope's capture and fission (if applicable) cross-sections\&. \fP"
.ti -1c
.RI "void \fBgenerateTotalXS\fP (float start_energy, float end_energy, int num_energies)"
.br
.RI "\fIComputes the microscopic total cross-section from the isotope's capture, elastic scatter and fission (if applicable) cross-sections\&. \fP"
.ti -1c
.RI "void \fBinitializeThermalScattering\fP (float start_energy, float end_energy, int num_bins, int num_distributions)"
.br
.RI "\fIThis method initializes the probability distributions for thermal scattering\&. \fP"
.ti -1c
.RI "float \fBthermalScatteringProb\fP (float E_prime_to_E, int dist_index)"
.br
.RI "\fIThis function computes the thermal scattering probability for a ratio of initial to final energies\&. \fP"
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "char * \fB_isotope_name\fP"
.br
.ti -1c
.RI "int \fB_A\fP"
.br
.ti -1c
.RI "float \fB_alpha\fP"
.br
.ti -1c
.RI "float \fB_eta\fP"
.br
.ti -1c
.RI "float \fB_rho\fP"
.br
.ti -1c
.RI "float \fB_AO\fP"
.br
.ti -1c
.RI "float \fB_T\fP"
.br
.ti -1c
.RI "float \fB_mu_avg\fP"
.br
.ti -1c
.RI "bool \fB_fissionable\fP"
.br
.ti -1c
.RI "bool \fB_rescaled\fP"
.br
.ti -1c
.RI "int \fB_num_elastic_xs\fP"
.br
.ti -1c
.RI "float * \fB_elastic_xs\fP"
.br
.ti -1c
.RI "float * \fB_elastic_xs_energies\fP"
.br
.ti -1c
.RI "bool \fB_elastic_rescaled\fP"
.br
.ti -1c
.RI "int \fB_num_capture_xs\fP"
.br
.ti -1c
.RI "float * \fB_capture_xs\fP"
.br
.ti -1c
.RI "float * \fB_capture_xs_energies\fP"
.br
.ti -1c
.RI "bool \fB_capture_rescaled\fP"
.br
.ti -1c
.RI "int \fB_num_fission_xs\fP"
.br
.ti -1c
.RI "float * \fB_fission_xs\fP"
.br
.ti -1c
.RI "float * \fB_fission_xs_energies\fP"
.br
.ti -1c
.RI "bool \fB_fission_rescaled\fP"
.br
.ti -1c
.RI "int \fB_num_absorb_xs\fP"
.br
.ti -1c
.RI "float * \fB_absorb_xs\fP"
.br
.ti -1c
.RI "float * \fB_absorb_xs_energies\fP"
.br
.ti -1c
.RI "int \fB_num_total_xs\fP"
.br
.ti -1c
.RI "float * \fB_total_xs\fP"
.br
.ti -1c
.RI "float * \fB_total_xs_energies\fP"
.br
.ti -1c
.RI "int \fB_num_energies\fP"
.br
.ti -1c
.RI "float \fB_start_lethargy\fP"
.br
.ti -1c
.RI "float \fB_end_lethargy\fP"
.br
.ti -1c
.RI "float \fB_delta_lethargy\fP"
.br
.ti -1c
.RI "bool \fB_use_thermal_scattering\fP"
.br
.ti -1c
.RI "float \fB_kB\fP"
.br
.ti -1c
.RI "int \fB_num_thermal_cdfs\fP"
.br
.ti -1c
.RI "int \fB_num_thermal_cdf_bins\fP"
.br
.ti -1c
.RI "float * \fB_thermal_dist\fP"
.br
.ti -1c
.RI "float ** \fB_thermal_cdfs\fP"
.br
.ti -1c
.RI "float * \fB_E_to_kT\fP"
.br
.ti -1c
.RI "float * \fB_Eprime_to_E\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
The \fBIsotope\fP represents a nuclide at some temperature\&. 

The \fBIsotope\fP class represents a nuclide and all of its properties which are relevant to reactor physics calculations\&. 
.SH "Constructor & Destructor Documentation"
.PP 
.SS "Isotope::Isotope (char *isotope_name)"

.PP
\fBIsotope\fP constructor\&. Searches the cross-section library for appropriately named files using the isotope name and loads the capture, scatter, and fission (if file is found) cross-sectoins\&. By default, the constructor rescales the cross-sections onto a uniform lethargy grid of 100,000 values between 1E-5 eV and 20 MeV\&. In addition, the constructor creates thermal scattering CDFs for the isotope at 300K by default\&. 
.SH "Member Function Documentation"
.PP 
.SS "\fBIsotope\fP * Isotope::clone ()"

.PP
This method clones a given \fBIsotope\fP class object by executing a deep copy of all of the \fBIsotope\fP's class attributes and giving them to a new \fBIsotope\fP class object\&. \fBReturns:\fP
.RS 4
a pointer to the new cloned \fBIsotope\fP class object 
.RE
.PP

.SS "void Isotope::collideNeutron (\fBneutron\fP *neutron)"

.PP
For a given energy, this method samples a collision type, updates the neutron's outgoing energy and kills the neutron if it sampled absorption or leakage\&. \fBParameters:\fP
.RS 4
\fIneutron\fP the neutron struct of interest 
.RE
.PP

.SS "void Isotope::generateAbsorptionXS (floatstart_energy, floatend_energy, intnum_energies)\fC [private]\fP"

.PP
Computes the microscopic absorption cross-section from the isotope's capture and fission (if applicable) cross-sections\&. This class method computes the absorption cross-section on a uniform lethargy grid\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstart_energy\fP the highest lethargy value in the grid 
.br
\fIend_energy\fP the lowest lethargy value in the grid 
.br
\fInum_energies\fP the number of energies represented in the grid 
.RE
.PP

.SS "void Isotope::generateTotalXS (floatstart_energy, floatend_energy, intnum_energies)\fC [private]\fP"

.PP
Computes the microscopic total cross-section from the isotope's capture, elastic scatter and fission (if applicable) cross-sections\&. This class method computes the total cross-section on a uniform lethargy grid\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstart_energy\fP the highest lethargy value in the grid 
.br
\fIend_energy\fP the lowest lethargy value in the grid 
.br
\fInum_energies\fP the number of energies represented in the grid 
.RE
.PP

.SS "int Isotope::getA () const"

.PP
Returns the atomic number of this isotope\&. \fBReturns:\fP
.RS 4
the atomic number 
.RE
.PP

.SS "float Isotope::getAbsorptionXS (floatenergy) const"

.PP
Returns the microscopic absorption cross-section value for some energy\&. Uses linear interpolation to compute the cross-section at a a certain energy (eV)\&. 
.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic absorption cross-section 
.RE
.PP

.SS "float Isotope::getAbsorptionXS (intenergy_index) const"

.PP
Returns the microscopic absorption cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fBParameters:\fP
.RS 4
\fIenergy_index\fP the index into the energy array 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic absorption cross-section 
.RE
.PP

.SS "float Isotope::getAlpha () const"

.PP
Returns the alpha $ \alpha = ((A-1)/(A+1))^2 $ values for this isotope\&. \fBReturns:\fP
.RS 4
alpha 
.RE
.PP

.SS "float Isotope::getCaptureXS (floatenergy) const"

.PP
Returns the microscopic capture cross-section value for some energy\&. Uses linear interpolation to compute the cross-section at a a certain energy (eV)\&. 
.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic absorption cross-section 
.RE
.PP

.SS "float Isotope::getCaptureXS (intenergy_index) const"

.PP
Returns the microscopic capture cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fBParameters:\fP
.RS 4
\fIenergy_index\fP the index into the energy array 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic capture cross-section 
.RE
.PP

.SS "float Isotope::getDistanceTraveled (\fBneutron\fP *neutron)"

.PP
For a given neutron, this method samples a distance traveled to next collision\&. \fBParameters:\fP
.RS 4
\fIneutron\fP the neutron struct of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the distance traveled 
.RE
.PP

.SS "float Isotope::getElasticXS (floatenergy) const"

.PP
Returns the microscopic elastic scattering cross-section value for some energy\&. Uses linear interpolation to compute the cross-section at a a certain energy (eV)\&. 
.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic elastic scattering cross-section 
.RE
.PP

.SS "float Isotope::getElasticXS (intenergy_index) const"

.PP
Returns the microscopic elastic cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fBParameters:\fP
.RS 4
\fIenergy_index\fP the index into the energy array 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic elastic cross-section 
.RE
.PP

.SS "int Isotope::getEnergyGridIndex (floatenergy) const\fC [inline]\fP"

.PP
This method returns the index for a certain energy (eV) into the \fBIsotope\fP's uniform lethargy grid\&. The index computed is that of nearest energy less than or equal to the input energy\&. 
.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the index into the uniform lethargy grid 
.RE
.PP

.SS "float Isotope::getFissionXS (floatenergy) const"

.PP
Returns the microscopic fission cross-section value for some energy\&. Uses linear interpolation to compute the cross-section at a a certain energy (eV)\&. 
.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic fission cross-section 
.RE
.PP

.SS "float Isotope::getFissionXS (intenergy_index) const"

.PP
Returns the microscopic fission cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fBParameters:\fP
.RS 4
\fIenergy_index\fP the index into the energy array 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic fission cross-section 
.RE
.PP

.SS "char * Isotope::getIsotopeName () const"

.PP
Returns the name of the of isotope\&. \fBReturns:\fP
.RS 4
character array with name of isotope 
.RE
.PP

.SS "float Isotope::getMuAverage () const"

.PP
Return the average value of the cosine of the scattering angle\&. Returns the average value of cosine of theta for this isotope in a scattering collision: $ \left<\mu\right> = \frac{2}{3A} $ 
.PP
\fBReturns:\fP
.RS 4
the average for mu 
.RE
.PP

.SS "int Isotope::getNumThermalCDFBins ()"

.PP
Returns the number of energy bins for each thermal scattering CDF\&. \fBReturns:\fP
.RS 4
the number of energy bins per thermal scattering CDF 
.RE
.PP

.SS "int Isotope::getNumThermalCDFs ()"

.PP
Returns the number of thermal scattering CDFs\&. \fBReturns:\fP
.RS 4
the number of thermal scattering CDFs 
.RE
.PP

.SS "int Isotope::getNumXSEnergies (char *xs_type) const"

.PP
Returns the number of energies for a particular cross-section type\&. Returns the number of energies for 'capture', 'elastic', 'fission', 'fission' and 'absorption cross-section types\&. 
.PP
\fBParameters:\fP
.RS 4
\fIxs_type\fP a character array for the xs_type 
.RE
.PP
\fBReturns:\fP
.RS 4
.RE
.PP

.SS "float Isotope::getTemperature () const"

.PP
Return the temperature in degrees Kelvin for this isotope\&. \fBReturns:\fP
.RS 4
the temperature of this isotope 
.RE
.PP

.SS "float Isotope::getThermalScatteringEnergy (floatenergy)"

.PP
For a given neutron energy (eV) in a scattering collision, this function returns the outgoing energy in eV, $ E' $, for the collision based on its thermal scattering distributions\&. \fBParameters:\fP
.RS 4
\fIenergy\fP the energy of the neutron of interest (eV) 
.RE
.PP
\fBReturns:\fP
.RS 4
the outgoing energy (eV) 
.RE
.PP

.SS "float Isotope::getTotalXS (floatenergy) const"

.PP
Returns the microscopic total cross-section value for some energy\&. Uses linear interpolation to compute the cross-section at a a certain energy (eV)\&. 
.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic total cross-section 
.RE
.PP

.SS "float Isotope::getTotalXS (intenergy_index) const"

.PP
Returns the microscopic total cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. \fBParameters:\fP
.RS 4
\fIenergy_index\fP the index into the energy array 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic total cross-section 
.RE
.PP

.SS "float Isotope::getTransportXS (intenergy_index) const"

.PP
Returns the microscopic transport cross-section value for a certain energy index in this isotope's rescaled uniform lethargy grid of cross-section data\&. The transport cross-section corrects for anisotropic scattering using a linear approximation and is computed as follows:
.PP
$ \sigma_{tr} = \sigma_t - \left<\mu\right>\sigma_s $.PP
\fBParameters:\fP
.RS 4
\fIenergy_index\fP the index into the energy array 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic transport cross-section 
.RE
.PP

.SS "float Isotope::getTransportXS (floatenergy) const"

.PP
Returns the microscopic transport cross-section value for some energy\&. Uses linear interpolation to compute the cross-section at a a certain energy (eV)\&. The transport cross-section corrects for anisotropic scattering and is computed as follows:
.PP
$ \sigma_{tr} = \sigma_t - \left<\mu\right>\sigma_s $.PP
\fBParameters:\fP
.RS 4
\fIenergy\fP the energy (eV) of interest 
.RE
.PP
\fBReturns:\fP
.RS 4
the microscopic transport cross-section 
.RE
.PP

.SS "void Isotope::initializeThermalScattering (floatstart_energy, floatend_energy, intnum_bins, intnum_distributions)\fC [private]\fP"

.PP
This method initializes the probability distributions for thermal scattering\&. It takes in arguments for the starting energy and end energy (ratios of kT) and the number of distributions which it uses to generate logarithmically spaced energies for the distributions\&. 
.PP
\fBParameters:\fP
.RS 4
\fIstart_energy\fP the first distribution's energy (ratio of kT) 
.br
\fIend_energy\fP the final distribution's energy (ratio of kT) 
.br
\fInum_bins\fP the number of bins per distribution 
.br
\fInum_distributions\fP the number of scattering distributions 
.RE
.PP

.SS "bool Isotope::isFissionable () const"

.PP
Return whether this isotope is fissionable (true) or not (false) \fBReturns:\fP
.RS 4
whether this isotope is fissionable 
.RE
.PP

.SS "bool Isotope::isRescaled () const"

.PP
This method returns whether or not the \fBIsotope\fP's cross-sections have been rescaled to a uniform lethargy grid\&. \fBReturns:\fP
.RS 4
whether or not the cross-sections have been rescaled 
.RE
.PP

.SS "void Isotope::loadXS ()\fC [private]\fP"

.PP
Load the ENDF cross-section data from ASCII files into arrays for this isotope\&. This method finds the appropriate ENDF data files for the isotope in the PINSPEC cross-section library based on the user-defined name of the isotope\&. If the appropriate files are not found the method will return an exception\&. If only capture and elastic scattering cross-section data files are discovered in the cross-section library then the isotope is not fissionable; otherwise if a fission cross-section file is found then the isotope is fissionable\&. Finally, after all cross-sections are parsed in from data files, this method computes a total cross-section and an absorption cross-section and then rescales all cross-sections onto a uniform lethargy grid to allow for fast O(1) data lookup\&. 
.SS "void Isotope::loadXS (char *xs_type)"

.PP

SWIG/isotope.py
# This file was automatically generated by SWIG (http://www.swig.org).
# Version 2.0.8
#
# Do not make changes to this file unless you know what you are doing--modify
# the SWIG interface file instead.



from sys import version_info
if version_info >= (2,6,0):
    def swig_import_helper():
        from os.path import dirname
        import imp
        fp = None
        try:
            fp, pathname, description = imp.find_module('_isotope', [dirname(__file__)])
        except ImportError:
            import _isotope
            return _isotope
        if fp is not None:
            try:
                _mod = imp.load_module('_isotope', fp, pathname, description)
            finally:
                fp.close()
            return _mod
    _isotope = swig_import_helper()
    del swig_import_helper
else:
    import _isotope
del version_info
try:
    _swig_property = property
except NameError:
    pass # Python < 2.2 doesn't have 'property'.
def _swig_setattr_nondynamic(self,class_type,name,value,static=1):
    if (name == "thisown"): return self.this.own(value)
    if (name == "this"):
        if type(value).__name__ == 'SwigPyObject':
            self.__dict__[name] = value
            return
    method = class_type.__swig_setmethods__.get(name,None)
    if method: return method(self,value)
    if (not static):
        self.__dict__[name] = value
    else:
        raise AttributeError("You cannot add attributes to %s" % self)

def _swig_setattr(self,class_type,name,value):
    return _swig_setattr_nondynamic(self,class_type,name,value,0)

def _swig_getattr(self,class_type,name):
    if (name == "thisown"): return self.this.own()
    method = class_type.__swig_getmethods__.get(name,None)
    if method: return method(self)
    raise AttributeError(name)

def _swig_repr(self):
    try: strthis = "proxy of " + self.this.__repr__()
    except: strthis = ""
    return "<%s.%s; %s >" % (self.__class__.__module__, self.__class__.__name__, strthis,)

try:
    _object = object
    _newclass = 1
except AttributeError:
    class _object : pass
    _newclass = 0


ELASTIC = _isotope.ELASTIC
ABSORPTION = _isotope.ABSORPTION
CAPTURE = _isotope.CAPTURE
FISSION = _isotope.FISSION
TRANSPORT = _isotope.TRANSPORT
DIFFUSION = _isotope.DIFFUSION
LEAKAGE = _isotope.LEAKAGE
TOTAL = _isotope.TOTAL
ISOTROPIC_CM = _isotope.ISOTROPIC_CM
ISOTROPIC_LAB = _isotope.ISOTROPIC_LAB
class Isotope(_object):
    __swig_setmethods__ = {}
    __setattr__ = lambda self, name, value: _swig_setattr(self, Isotope, name, value)
    __swig_getmethods__ = {}
    __getattr__ = lambda self, name: _swig_getattr(self, Isotope, name)
    __repr__ = _swig_repr
    def __init__(self, *args): 
        this = _isotope.new_Isotope(*args)
        try: self.this.append(this)
        except: self.this = this
    __swig_destroy__ = _isotope.delete_Isotope
    __del__ = lambda self : None;
    def parseName(self): return _isotope.Isotope_parseName(self)
    def makeFissionable(self): return _isotope.Isotope_makeFissionable(self)
    def getIsotopeType(self): return _isotope.Isotope_getIsotopeType(self)
    def getA(self): return _isotope.Isotope_getA(self)
    def getAlpha(self): return _isotope.Isotope_getAlpha(self)
    def getN(self): return _isotope.Isotope_getN(self)
    def getAO(self): return _isotope.Isotope_getAO(self)
    def getTemperature(self): return _isotope.Isotope_getTemperature(self)
    def getMuAverage(self): return _isotope.Isotope_getMuAverage(self)
    def isFissionable(self): return _isotope.Isotope_isFissionable(self)
    def getElasticXS(self, *args): return _isotope.Isotope_getElasticXS(self, *args)
    def getElasticAngleType(self): return _isotope.Isotope_getElasticAngleType(self)
    def getAbsorptionXS(self, *args): return _isotope.Isotope_getAbsorptionXS(self, *args)
    def getCaptureXS(self, *args): return _isotope.Isotope_getCaptureXS(self, *args)
    def getFissionXS(self, *args): return _isotope.Isotope_getFissionXS(self, *args)
    def getTotalXS(self, *args): return _isotope.Isotope_getTotalXS(self, *args)
    def getTransportXS(self, *args): return _isotope.Isotope_getTransportXS(self, *args)
    def usesThermalScattering(self): return _isotope.Isotope_usesThermalScattering(self)
    def setIsotopeType(self, *args): return _isotope.Isotope_setIsotopeType(self, *args)
    def setA(self, *args): return _isotope.Isotope_setA(self, *args)
    def setAO(self, *args): return _isotope.Isotope_setAO(self, *args)
    def setN(self, *args): return _isotope.Isotope_setN(self, *args)
    def setTemperature(self, *args): return _isotope.Isotope_setTemperature(self, *args)
    def loadXS(self, *args): return _isotope.Isotope_loadXS(self, *args)
    def setElasticXS(self, *args): return _isotope.Isotope_setElasticXS(self, *args)
    def setElasticAngleType(self, *args): return _isotope.Isotope_setElasticAngleType(self, *args)
    def setAbsorptionXS(self, *args): return _isotope.Isotope_setAbsorptionXS(self, *args)
    def setCaptureXS(self, *args): return _isotope.Isotope_setCaptureXS(self, *args)
    def setFissionXS(self, *args): return _isotope.Isotope_setFissionXS(self, *args)
    def generateCaptureXS(self): return _isotope.Isotope_generateCaptureXS(self)
    def rescaleXS(self, *args): return _isotope.Isotope_rescaleXS(self, *args)
    def clone(self): return _isotope.Isotope_clone(self)
    def getCollisionType(self, *args): return _isotope.Isotope_getCollisionType(self, *args)
    def collideNeutron(self, *args): return _isotope.Isotope_collideNeutron(self, *args)
    def getThermalScatteringEnergy(self, *args): return _isotope.Isotope_getThermalScatteringEnergy(self, *args)
    def initializeThermalScattering(self, *args): return _isotope.Isotope_initializeThermalScattering(self, *args)
    def thermalScatteringProb(self, *args): return _isotope.Isotope_thermalScatteringProb(self, *args)
    def addTally(self, *args): return _isotope.Isotope_addTally(self, *args)
    def clearTallies(self): return _isotope.Isotope_clearTallies(self)
Isotope_swigregister = _isotope.Isotope_swigregister
Isotope_swigregister(Isotope)

# This file is compatible with both classic and new-style classes.


